subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/ArtifactsTestListener.kt

```diff
@@ -65,13 +65,13 @@ internal class ArtifactsTestListener(
             Passed,
             is Failed.InRun -> {
                 val artifacts = testArtifactsDir.flatMap { dir ->
-
-                    // last /. means to adb to copy recursively, and do not copy the last
-                    // example:
-                    //  - from: /sdcard/Android/someDir/ to: /xx ; will copy to /xx/someDir/ and not recursive
-                    //  - from: /sdcard/android/someDir/. to: /xx ; will copy to /xx and recursive
-                    // todo move this knowledge under adb layer
-                    device.pull(from = dir.toPath() / ".", to = tempDirectory)
+                    device.pullDir(
+                        deviceDir = dir.toPath(),
+                        hostDir = tempDirectory,
+                        validator = ReportAwarePullValidator(
```

Comment:
I think that clients do not need to know about `validator`
validator looks like device implementation details